happi/containers.py

```diff
@@ -1,330 +1,30 @@
-    kwargs = deepcopy(Device.kwargs)
-    kwargs.default.update({'mono_line': '{{mono_line}}',
-                           'main_line': '{{beamline}}'})
-
-
-class MovableStand(Device):
-    """
-    This class stores information about stands that move, like XPP's hand-crank
-    that moves SB2 and SB3 from the PINK to XPP lines and back. There is no
-    need to instantiate one of these for static stands.
-
-    Attributes
-    ----------
-    prefix : str
-        If there is a single PV with the stand's location, this should be the
-        base PV. In general, these devices will actually have multiple PVs
-        with binary outputs that have yes/no on the stand being in each
-        position. In these cases we pick the common prefix of these PVs.
-
-    stand : list
-        List of stands affected by table movement.
-    """
-    stand = copy(Device.stand)
-    stand.enforce = list
-    system = copy(Device.system)
-    system.default = 'changeover'
-
-
-class Motor(Device):
-    """
-    A Generic EpicsMotor
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Motor'
-    system = copy(Device.system)
-    system.default = 'motion'
-
-
-class AreaDetector(Device):
-    """
-    A Generic EpicsCamera
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.PCDSDetector'
-    system = copy(Device.system)
-    system.default = 'camera'
-
-
-class Acromag(Device):
-    """
-    A Generic class for Acromag
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Acromag'
-    system = copy(Device.system)
-    system.default = 'acromag'
-
-
-class Trigger(Device):
-    """
-    A Generic class for Controls Triggers
-    """
-    device_class = copy(Device.device_class)
-    device_class.default = 'pcdsdevices.device_types.Trigger'
-    system = copy(Device.system)
-    system.default = 'timing'
+import logging
+import inspect
+import entrypoints
+
+from .item import HappiItem
+
+logger = logging.getLogger(__name__)
+
+_entries = entrypoints.get_group_all('happi.containers')
+
+registry = set()
+
+for entry in _entries:
+    try:
+        obj = entry.load()
+    except Exception:
+        logger.exception("Failed to load happi.containers entry: %s", entry)
+        continue
+    if inspect.isclass(obj) and HappiItem in obj.mro():
+        registry.add(obj)
+    elif inspect.ismodule(obj):
+        registry.update([var for _, var in obj.__dict__.items()
+                         # Only classes
+                         if inspect.isclass(var)
+                         # Must inherit from HappiItem at some point
+                         and HappiItem in var.mro()
+                         # Avoid happi internal classes due to imports
+                         and not var.__module__.startswith('happi.')])
+
+        locals().update({c.__name__: c for c in registry})
```

Comment:
Indentation right here (seems like this should happen at the end after the registry is built)?

I think you're trying to get `happi.containers.<classname>` working: `globals()` seems more correct but may do the same here. We're also risking name collisions in the namespace as this strips off the `__module__` prefix, hmm...

Reply:
Indentation solved with https://github.com/pcdshub/happi/pull/126/commits/5c39ffbb8d283184a295b6080631c60f033c681e. Sorry that I forgot to push before opening the PR.

Reply:
About the globals()... I'm not sure.

Reply:
ok, now I understand... that is correct, we inject the class name into the `happi.containers` namespace it would fail if we have 2 containers with the same name but different full qualified name.